php-checks/src/main/java/org/sonar/php/checks/security/CORSPolicyCheck.java

```diff
@@ -0,0 +1,115 @@
+import static org.sonar.plugins.php.api.tree.Tree.Kind.REGULAR_STRING_LITERAL;
+
+@Rule(key = "S5122")
+public class CORSPolicyCheck extends PHPVisitorCheck {
+
+  private static final String MESSAGE = "Make sure this permissive CORS policy is safe here.";
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree functionCallTree) {
+    FunctionCallHelper functionCallHelper = FunctionCallHelper.create(functionCallTree);
+
+    if ((functionCallHelper.isResponseConstuctorVulnerable()) || (functionCallHelper.isCoreHeaderVulnerable()) || (functionCallHelper.isSetOrHeaderVulnerable())) {
+      context().newIssue(this, functionCallTree, MESSAGE);
+    }
+    super.visitFunctionCall(functionCallTree);
+  }
+
+  private static class FunctionCallHelper {
+
+    private static final String HEADER = "header";
+
+    private FunctionCallTree functionCallTree;
+    private ExpressionTree callee;
+
+    private FunctionCallHelper(FunctionCallTree functionCallTree) {
+      this.functionCallTree = functionCallTree;
+      this.callee = functionCallTree.callee();
+    }
+
+    static FunctionCallHelper create(FunctionCallTree functionCallTree) {
+      return new FunctionCallHelper(functionCallTree);
+    }
+
+    private boolean isResponseConstuctorVulnerable() {
+      return isResponseConstructorFunctionCall() &&
+        argument(functionCallTree, "headers", 2)
+          .map(CallArgumentTree::value)
+          .filter(ArrayInitializerBracketTree.class::isInstance)
+          .map(ArrayInitializerBracketTree.class::cast)
+          .filter(a -> a.arrayPairs()
+            .stream()
+            .anyMatch(b -> isLiteralTreeEqualsTo(b.key(), "Access-Control-Allow-Origin") && isLiteralTreeEqualsTo(b.value(), "*")))
+          .isPresent();
+    }
+
+    private boolean isCoreHeaderVulnerable() {
+      return isCoreHeaderFunctionCall() &&
+        retrieveArgumentAndVerifyItIsEqualsTo(HEADER, 0, "Access-Control-Allow-Origin:*");
+    }
+
+    private boolean isSetOrHeaderVulnerable() {
+      return isSetOrHeaderFunctionCall() &&
+        retrieveArgumentAndVerifyItIsEqualsTo("key", 0, "Access-Control-Allow-Origin") &&
+        retrieveArgumentAndVerifyItIsEqualsTo("values", 1, "*");
+    }
+
+    private boolean retrieveArgumentAndVerifyItIsEqualsTo(String name, int position, String expected) {
+      Optional<CallArgumentTree> argumentTree = argument(functionCallTree, name, position);
+      return argumentTree.isPresent() && isLiteralTreeEqualsTo(argumentTree.get().value(), expected);
+    }
+
+    private static boolean isLiteralTreeEqualsTo(ExpressionTree tree, String expected) {
+      return tree.is(REGULAR_STRING_LITERAL) && expected.equalsIgnoreCase(trimQuotes(((LiteralTree) tree).value().replaceAll("\\s", "")));
+    }
+
+    private boolean isResponseConstructorFunctionCall() {
+      return callee instanceof ClassNamespaceNameTreeImpl && "Response".equalsIgnoreCase(nameOf(callee));
```

Comment:
`ClassNamespaceNameTreeImpl` is not intended to be used as an API.
I would prefer something like:
```
return callee.is(NAMESPACE_NAME) && responseClasses.contains(Symbols.getClass((NamespaceNameTree) callee).qualifiedName());
```
That would use an API which is more suited to what you want to check. You wouldn't have to care about case-sensitivity and it would catch cases like:
```
use Illuminate\Http\Response as MyResponse;
new Myresponse(...);
```

Reply:
nice, and fixed!